### ISSUE TITLE
Consider removing the underline on the post titles to improve readability.

### ISSUE THREAD

**Issue by marctison75** [WordPress/twentytwentyone#390]
As a person who has dyslexia, underline makes it harder for me to read. For the posts page I find it difficult to read the titles. I’ve checked the Figma files, and there is no underline for the post titles. This ticket #110 has been closed, but I’m not sure if there was a consensus.

**Comment by melchoyce** [comment]
We'll need a definitive answer from the a11y team about whether or not the site title and post titles need to have explicit link indicators.

**Comment by sarahricker** [comment]
Hey @marcduluc and @melchoyce! 

In our new joint office hours, the Design and Accessibility teams discussed ways to help make the titles more readable, while still meeting the WCAG AAA guidelines to the best of our ability. However, different disabilities can create the need for a balancing act regarding overlapping issues. For instance, it is important to have a visual cue on the resting state for all links. 

**So far, our recommendations are to:**

- [ ] **Remove the link on the post title, if possible**
It would be ideal to avoid duplicate links to the same page and this would solve the need for underlined titles all together. 

- [x] **Ensure use of `text-decoration-skip-ink` across all links for the site**
This should be on by default in most browsers, but just in case, let's double check. The `text-decoration-skip-ink` feature can [improve readability for dyslexic users](https://medium.com/@iamhiwelo/improving-text-readability-for-dyslexic-users-with-skip-ink-underlines-bf52a2f3426b), though it is not a 100% guarantee that it will work for everyone.

 Can we try the above and see if that helps? If not, we can do a deeper dive if needed.

**Comment by melchoyce** [comment]
Hey @sarahricker! 

> It would be ideal to avoid duplicate links to the same page and this would solve the need for underlined titles all together.

I think @marcduluc means the archives, not single posts:

![image](https://user-images.githubusercontent.com/2846578/95917121-bd1d1080-0d77-11eb-98a9-7d0e17cd8964.png)

Single posts don't link to their own titles.

>  Ensure use of `text-decoration-skip-ink` across all links for the site

It looks like we don't have this explicitly defined, but I'm still seeing the gap on Chrome, Firefox, and Safari. What browser are you testing in?

Thanks for reviewing!

**Comment by sarahricker** [comment]
Ahh, thanks @melchoyce !

Maybe a button below the excerpt, or other discernable treatment on the archive post title links (like ellipses) would help? I'm not opposed to removing the underline, so long as there is still an obvious link (more than color only). 



Re `text-decoration-skip-ink`  I see it as well,  but thinking we should define it just in case older/ less common browsers don't use it by default.  :)

**Comment by melchoyce** [comment]
Thanks @sarahricker!

**Comment by aristath** [comment]
@sarahricker following up from a [Slack discussion](https://wordpress.slack.com/archives/C02RP4X03/p1602645132154100) I saw earlier today, could we please have a clear decision for this?

> Maybe a button below the excerpt, or other discernable treatment on the archive post title links (like ellipses) would help? I'm not opposed to removing the underline, so long as there is still an obvious link (more than color only).

Right now, post-archives look like this:

![Screenshot_2020-10-14 Page 2 – My Blog](https://user-images.githubusercontent.com/588688/95963511-716a7580-0e10-11eb-915f-70a85f33fd8d.png)

So there is a "Read More" link at the end of excerpts. Post-titles are focusable by tabbing the keyboard, and the read-more link has the default (underlined) style for links.

If we remove underlines from titles (pending PR: #461), then posts in the archive will look like this:

![Peek 2020-10-14 11-31](https://user-images.githubusercontent.com/588688/95963916-ee95ea80-0e10-11eb-8275-00011d78884b.gif)

In that gif I included normal, hover & focus just to get an idea of how it would look.

**Comment by carolinan** [comment]
@aristath  Did you mean _If we remove underlines from titles_?

There are 3 links to the post here: Title, continue reading, and date. 

If logged in, the post title is repeated 3 times:
Post title, continue reading post title, edit post title

The exception is the aside that does not show the title.

There is a filter for the title so there is always a title text even if the user does not enter one.
<del>Because of that, the link on the date is not needed.</del>  Scratch that, it is needed because the aside does not have a read more link or a title. Hmmm. Maybe only link the date for the aside format?

**Comment by aristath** [comment]
> @aristath Did you mean If we remove underlines from titles?

:facepalm: I did, thank you! Fixed the typo :+1:

**Comment by enriquesanchez** [comment]
We need some kind of visual treatment for links in order to comply with WCAG guidelines, which is a goal of this theme. The most common and obvious method is the default underline style.

@marcduluc Do you find that the skip-ink effect in underlines helps with readability?

<img width="403" alt="Screen Shot 2020-10-14 at 9 45 23" src="https://user-images.githubusercontent.com/3276087/96005375-fe5a0280-0e01-11eb-8cc6-ca0e3bc34edd.png">

We can also try a few different variations of underline, with properties such as `text-underline-offset` and `text-decoration-thickness` (though they are not supported in all browser versions):

<img width="478" alt="Screen Shot 2020-10-14 at 9 48 07" src="https://user-images.githubusercontent.com/3276087/96005861-804a2b80-0e02-11eb-9ac1-06e2e8b47fdc.png">

@marcduluc Do you find any of the above to be easier to read?

**Comment by afercia** [comment]
> ~Because of that, the link on the date is not needed.~ Scratch that, it is needed because the aside does not have a read more link or a title. Hmmm. Maybe only link the date for the aside format?

Will create a new issue for this. Rationale: a link text like, for example, "October 14, 2020" that links to the post permalink doesn't make much sense for all users (visually and semantically) and also for SEO. Also, it would be nice to reduce the amount of redundant links in the archive pages and find a workaround for the Aside (and the Status).

**Comment by afercia** [comment]
An in-depth analysis on link underlines:

On Link Underlines
https://adrianroselli.com/2016/06/on-link-underlines.html

It includes references and considerations about dyslexia as well. To my understanding, the most important part is to make sure the underline doesn't have a big impact ("overlaps") on the ability to recognize the characters shape as in: the underline should never intersect the typeface descenders. The CSS property `text-decoration-skip-ink` goes in the right direction. The article author also suggests to explore underlines with a lighter color using `text-decoration-color`, as long as the contrast ratio is sufficient.

**Comment by marctison75** [comment]
@enriquesanchez Yes the skip-ink effect in underlines helps with readability. But for me, underline titles with skip-ink are more difficult to read  than titles with no underline. As  blog post pages have many titles, it would be a bit complicated for me to browse the page.

But I understand that there are also other requirements to take into consideration 🙂

**Comment by kjellr** [comment]
Based on the conversation above, it seems to me that keeping an underline present is our best solution. 

I'd advise against using a `border-bottom` here — we've removed that from all other links already. And since we cannot control the `text-underline-thickness` in Chrome, using a `border-bottom` for just the post title would introduce a weird situation where the post title underline would be less thick than other underlines on the page: 

<img width="667" alt="Screen Shot 2020-10-15 at 8 44 23 AM" src="https://user-images.githubusercontent.com/1202812/96128101-ee562780-0ec2-11eb-8e25-99475d71619e.png">

Given that, I think our best bet is to just modify the offset of the underline, so that it's not as mixed up with the text at a glance. Yes, this will have no effect in Chrome, but [it appears](https://chromestatus.com/feature/5747636764147712) that all these text decoration modifications are likely to arrive in Chrome at some point, so I'm not sure it makes sense to work around them too much at this point. 

I opened #502 to try this out.  

Current (Firefox/Safari): 
<img width="500" alt="Screen Shot 2020-10-15 at 8 34 42 AM" src="https://user-images.githubusercontent.com/1202812/96128556-83592080-0ec3-11eb-8c11-27a0ca924dfd.png">

Updated:
<img width="527" alt="Screen Shot 2020-10-15 at 8 34 29 AM" src="https://user-images.githubusercontent.com/1202812/96128570-86541100-0ec3-11eb-911f-97693539a04e.png">

**Comment by pattonwebz** [comment]
I wanted to re-open this issue here because, while wo do have _a working solution_ for some instances it's not ideal and the [support level that it has is really quite small](https://caniuse.com/?search=text-underline). Basically current support is only safari and firefox for desktop (neither support it on mobile).

The merged solution is only suitable for less than 10% of total users (consider collectively their [desktop market share is just 10.7%](https://caniuse.com/?search=text-underline) and if we added mobile users to that it is really a much smaller share of total users).

**Comment by enriquesanchez** [comment]
@pattonwebz I think that keeping the underline is the best option we have now. There's not a one size fits all solution here. 

We can improve on it where possible, with things such as skip-ink, text-underline-thickness, and text-underline-offset and let users adjust after that to their needs.

I highly recommend the article shared by @afercia above: https://adrianroselli.com/2016/06/on-link-underlines.html

It's highly informative and explores many different approaches, and the underline ends up being the most solid solution.

**Comment by pattonwebz** [comment]
@enriquesanchez I am inclined to agree we have the best solution we can have here however I reopened the issue to give a little extra time to explore more and maybe find another solution. After my own very quick explore so far I haven't found anything that workws better but I'd still like to keep this open a little longer to give a bit more time (and so that I don't forget to circle back :smile: ).

**Comment by joyously** [comment]
I noticed that the link looks a lot like inserted text (using `<ins>`).
![link-text](https://user-images.githubusercontent.com/7957680/96313501-4d0ac680-0fd3-11eb-86bc-e8d7bcfe0cbe.jpg)

**Comment by sarahricker** [comment]
Discussed again in a11y + Design office hours (see [slack conversation](https://wordpress.slack.com/archives/C02RP4X03/p1603209410367200)).

Those present agreed that the underline is the most solid solution, but also understand the desire to finesse the approach some more. @joyously your screenshot is another good example, probably just needs a small tweak. 😄 

Given that today is Beta 1, I don't think much more will make it in. However, will keep this issue open for future theme updates.

**Comment by pattonwebz** [comment]
I spent a bit of time looking at this and cannot find any better solutions. Closing for now as I don't think any better solutions.
